From: Martin Schwidefsky <schwidefsky@de.ibm.com>
Subject: s390/3270: add missing tty_kref_put
Patch-mainline: v4.7-rc1
Git-commit: 5ff04fe5659f78c0450cc6858b12bba9e4049578
References: bnc#980892, LTC#141736

Description:  kernel: fix disconnect/reconnect of 3270 terminals
Symptom:      The disconnect and reconnect of a 3270 terminal is not
              working as expected, the terminal session is not hung
              up and the new size of the reconnected terminal is not
              detected.
Problem:      The 3270 driver does not have support for disconnected
              terminals.
Solution:     Add support for disconnected 3270 terminals.
Reproduction: Open a 3270 terminal, login on the tty, disconnected the
              terminal and reconnect with a terminal of different size.

Upstream-Description:

              s390/3270: add missing tty_kref_put

              The tty3270_resize_work functions acquires a reference to the tty
              to call tty_do_resize but fails to release the reference again.
              Add the missing tty_kref_put.

              Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>


Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>
Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/s390/char/tty3270.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/s390/char/tty3270.c
+++ b/drivers/s390/char/tty3270.c
@@ -838,6 +838,7 @@ static void tty3270_resize_work(struct w
 	ws.ws_row = tp->view.rows - 2;
 	ws.ws_col = tp->view.cols;
 	tty_do_resize(tty, &ws);
+	tty_kref_put(tty);
 }
 
 static void
